libraries/classes/Display/Results.php

```diff
@@ -3966,14 +3966,15 @@ public function setConfigParamsForDisplayTable()
             $query['repeat_cells'] = $GLOBALS['cfg']['RepeatCells'];
         }
 
+        $sessionMaxRows = $_GET['session_max_rows'] ?? $_POST['session_max_rows'];
         // as this is a form value, the type is always string so we cannot
         // use Core::isValid($_POST['session_max_rows'], 'integer')
-        if (Core::isValid($_POST['session_max_rows'], 'numeric')) {
-            $query['max_rows'] = (int) $_POST['session_max_rows'];
-            unset($_POST['session_max_rows']);
-        } elseif ($_POST['session_max_rows'] == self::ALL_ROWS) {
+        if (Core::isValid($sessionMaxRows, 'numeric')) {
+            $query['max_rows'] = (int) $sessionMaxRows;
+            unset($_REQUEST['session_max_rows']);
```

Comment:
Could you unset get and post?
```suggestion
            unset($_GET['session_max_rows'], $_POST['... );
```